docs/services/rook-services-guide.md

```diff
@@ -0,0 +1,652 @@
+
+## Set Up Resource Classes
+
+In order to dynamically provision resources, we need to create resources classes that contain details about how the resources should be provisioned. For Yugastore, we will need resource classes that are capable of fulfilling a `KubernetesCluster` claim and a `PostgreSQLInstance` claim:
+
+```bash
+cat > classes.yaml <<EOF
+apiVersion: compute.gcp.crossplane.io/v1alpha2
+kind: GKEClusterClass
+metadata:
+  name: standard-gke
+  labels:
+    app: yugastore
+specTemplate:
+  machineType: n1-standard-1
+  numNodes: 1
+  zone: us-central1-b
+  providerRef:
+    name: gcp-provider
+  reclaimPolicy: Delete
+---
+apiVersion: database.rook.crossplane.io/v1alpha1
+kind: YugabyteClusterClass
+metadata:
+  name: yuga-cluster
+specTemplate:
+  providerRef:
+    name: yugastore-k8s-provider
+  reclaimPolicy: Delete
+  forProvider:
+    name: hello-ybdb-cluster
+    namespace: rook-yugabytedb
+    master:
+      # Replica count for Master.
+      replicas: 3
+      # Mentioning network ports is optional. If some or all ports are not specified, then they will be defaulted to below-mentioned values, except for tserver-ui.
+      network:
+        ports:
+          - name: yb-master-ui
+            port: 7000          # default value
+          - name: yb-master-rpc
+            port: 7100          # default value
+      # Volume claim template for Master
+      volumeClaimTemplate:
+        metadata:
+          name: datadir
+        spec:
+          accessModes: [ "ReadWriteOnce" ]
+          resources:
+            requests:
+              storage: 1Gi
+          storageClassName: standard
+    tserver:
+      # Replica count for TServer
+      replicas: 3
+      # Mentioning network ports is optional. If some or all ports are not specified, then they will be defaulted to below-mentioned values, except for tserver-ui.
+      # For tserver-ui a cluster ip service will be created if the yb-tserver-ui port is explicitly mentioned. If it is not specified, only StatefulSet & headless service will be created for TServer. TServer ClusterIP service creation will be skipped. Whereas for Master, all 3 kubernetes objects will always be created.
+      network:
+        ports:
+          - name: yb-tserver-ui
+            port: 9000
+          - name: yb-tserver-rpc
+            port: 9100          # default value
+          - name: ycql
+            port: 9042          # default value
+          - name: yedis
+            port: 6379          # default value
+          - name: ysql
+            port: 5433          # default value
+      # Volume claim template for TServer
+      volumeClaimTemplate:
+        metadata:
+          name: datadir
+        spec:
+          accessModes: [ "ReadWriteOnce" ]
+          resources:
+            requests:
+              storage: 1Gi
+          storageClassName: standard
+EOF
+
+kubectl apply -f classes.yaml
+```
+
+The `GKEClusterClass` is relatively straightforward in that it configures a
```

Comment:
This section is pretty wordy and in the weeds, which I am not sure is exactly what we are looking for here. However, because `stack-rook` is different than other infra stacks it felt like some extra explanation was added. I tend to dive too deep in explanations so would love tone / content feedback from @prasek / @jbw976 here